sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/models/AzureJsonHelper.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.monitor.opentelemetry.exporter.implementation.models;
+
+import com.azure.json.JsonWriter;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public final class AzureJsonHelper {
+
+    // this method is replacing JsonWriter.writeMapField(), which would generate json attributes with null value.
+    // This method is used to write a map to a json writer.
+    // The map is written as a json object with the key as the field name and the value as the field value.
+    // The value is written using the writeValueCallback.
+    public static <T> void writeMap(Map<String, T> map, String fieldName, JsonWriter jsonWriter, Consumer<T> writeValueCallback) throws IOException {
```

Comment:
e.g. (from above) we could change `Consumer` here to be our own functional interface whose method is declared to throw IOException

Reply:
let me try that

Reply:
are you suggesting something like this 

```java
 interface ThrowingConsumer<T> {
    void accept(T t) throws Exception;
}
```

this is a temporary workaround; we'll remove it once azure-json supports [it](https://github.com/Azure/azure-sdk-for-java/issues/41269)

Reply:
yes 👍